lib/active_record_doctor/detectors/mismatched_foreign_key_type.rb

```diff
@@ -31,6 +31,7 @@ def detect
 
             next if ignored?("#{table}.#{from_column.name}", config(:ignore_columns))
 
+            # TODO: Add support for composite primary keys when https://github.com/rails/rails/pull/47637 is merged.
```

Comment:
That PR adds a proper way to define multi-column foreign keys and getting their metadata, which we can not get correctly currently.

Reply:
That was merged, so needs an update before merging this PR.